### PR TITLE
Proposed feature, new  destination: Azure blob storage

### PR DESCRIPTION
**Pull request by RyanCPeters**
## NOTICE: This PR is not finalized...
Rather than put off sharing the proposed new feature till after this holiday weekend, I'm sharing it now with the provisor that I haven't fully reviewed the repo's contributing guidelines. I'll update and format this PR after the weekend.

This PR defines a new destination: `class AzureBlob(BaseDestination)`. It also defines multiple new situationally specific Exception classes with `DestinationError` as their root base class. 

This PR also includes test files for validating expected behavior from `AzureBlob` instances built on the `Unittest` test-runner. To support these test scripts, there has been a new recipe added to the project's root Makefile, along with comments added to the `tox.ini` file that give an example of how one may define cumulative coverage reports when running both `pytest` and `unittest` test-runners in separate commands.

### CODE REVIEW

**Comment by RyanCPeters** [comment]
Hello @akuzminsky, @ernstae suggested I spin up my work for adding Azure blob storage as a new destination type to your twindb/backup repo. I meant to get this PR in to you a few days ago, but I haven't looked at the project in many months so it took me a moment to validate that everything was still in good working order.

**Comment by akuzminsky** [comment]
Hi @RyanCPeters, 
Thank you for the pull request. I sent you a PR https://github.com/RyanCPeters/backup/pull/1 to fix pylint and black errors. I took this opportunity to reconfigure max line width to 120 characters, so there is a lot of noop changes. One real change though was lamda vs a function as per pylint request. Please feel free to review/merge and then I will merge this diff in the master.

As for the tests. There are two test suites in this repo - one [unit tests](https://github.com/twindb/backup/tree/master/tests/unit) that runs in github actions. It does not require external resources, all of them are mocked out with `mock` classes. Written for pytest.

There is another suite - [integration tests](https://github.com/twindb/backup/tree/master/tests/integration). They do require S3 and other external resources. I run them on my computer. Written also for pytest.

It seem to me that your tests would extend the integration tests suite. In principle, it doesn't matter what test framework the integration tests use. So I will try to run them as they're now.

A coverage test report shows how TwinDB Backup is covered with _unit_ tests. The report file has to be generated by one tool, so at some point I will add unit tests for the AzureBlob destination.

To summarize:
* Please review/merge my [PR](https://github.com/RyanCPeters/backup/pull/1).
* I will merge this PR and then will figure out unit/integration tests and coverage report.

**Comment by RyanCPeters** [comment]
Excellent, I'll look over your PR and get that merged asap. 

Regarding the unit tests: I didn't realize there were two groups, an oversight on my part sorry, but that structure makes a lot of sense. Yes, the tests I've made for the azure blob destination are certainly in the integration category. When I get a chance, in the future, I'll need to take a closer look at how to set up mock classes and proper pytest functions as a sort of regression test suite for this feature addition. I'd like to have a better understanding of that testing pattern.
